### DISCUSSION TITLE
Calling parent function from another parent funtion

### DISCUSSION THREAD

**Discussion by ctimmer** [micropython/micropython#12090] [Using MicroPython] [answered]
I am trying to call a parent class function (test2, overridden in subclass B) from another parent class function (test1). With the code below self.test2() in class A the function called is actually test2 in class B (see output). I can see why it works this way but is there a way to force a function call to use the parent function even if it is overridden in the child class (B)? Also class A should be able to be used by itself.

```py
#---- Parent
class A () :
    def __init__ (self) :
        pass
    def test1 (self) :
        print ("A test1")
        self.test2 ("from a")
        self.test3 ("from a")
    def test2 (self, mess) :
        print ("A test2", mess)
    def test3 (self, mess) :
        print ("A test3", mess)

class B (A) :
    def __init__ (self) :
        super().__init__ ()
    def test1 (self) :
        print ("B test1")
        super().test1 ()
    def test2 (self, mess) :
        print ("B test2", mess)

Z = B()
Z.test1 ()

### Output:
B test1
A test1       
B test2 from a   # Problem
A test3 from a
```
Thanks for the help,
Curt

**Comment by dlech** [comment, marked as answer]
Instead of `self.test2("from a")`, call `A.test2(self, "from a")`. This will always call the `test2` method defined in `A`.

**Reply by dlech** [reply]
Or even better IMHO, use composition instead of inheritance.

**Comment by ctimmer** [comment]
> Or even better IMHO, use composition instead of inheritance.

I considered this but it makes access to the class variables more difficult.
Thanks for the input
